Mathlib/NumberTheory/Ostrowski.lean

```diff
@@ -0,0 +1,64 @@
+/-
+Copyright (c) 2024 María Inés de Frutos-Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Kurniadi Angdinata, Fabrizio Barroero, Laura Capuano, Nirvana Coppola,
+María Inés de Frutos Fernández, Sam van Gool, Silvain Rideau-Kikuchi, Amos Turchet,
+Francesco Veneziano
+-/
+
+import Mathlib.Analysis.Normed.Field.Basic
+import Mathlib.Analysis.SpecialFunctions.Pow.Real
+import Mathlib.Analysis.Normed.Ring.Seminorm
+
+/-!
+# Ostrowski’s Theorem
+
+The goal of this file is to prove Ostrowski’s Theorem which gives a list of all the nontrivial
+absolute values on a number field up to equivalence.
+
+## References
+* https://kconrad.math.uconn.edu/blurbs/gradnumthy/ostrowskiQ.pdf
+* https://kconrad.math.uconn.edu/blurbs/gradnumthy/ostrowskinumbfield.pdf
+
+## Tags
+ring_norm, ostrowski
+-/
+
+noncomputable section
+namespace Rat.MulRingNorm
+open Int
+
+variable {f g : MulRingNorm ℚ}
+
+/-- Values of a multiplicative norm of the rationals are determined by the values on the
```

Comment:
The lemma below does not really say this: it says that if two norms coincide on `ℕ` then they coincide on `ℤ` and conversely.

Reply:
Don't know why I (or whoever wrote it, I don't remember anymore) wrote it like this, probably a distraction given by the similarity with the other comments. :)